backend/app/api/ws/chat_namespace.py

```
@@ -809,6 +811,18 @@ async def on_chat_send(self, sid: str, data: dict) -> dict:
                 "message_id": user_subtask.message_id if user_subtask else None,
             }
 
+        except HTTPException as e:
+            # Handle HTTPException specially to preserve structured error details
+            # This is crucial for errors like TASK_EXPIRED_RESTORABLE (409)
+            # where the detail is a dict that frontend needs to parse
+            logger.exception(f"[WS] chat:send HTTPException: {e}")
```

Ruff (0.14.14)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)